### PR TITLE
Update LLVM to 8066f22c4663

### PR DESCRIPTION
**Pull request by fabianschuiki**
This pulls in a few improvements that allow function-like ops to more easily insert arguments and results, among other things.

### CODE REVIEW

**Comment by fabianschuiki** [comment]
Going to merge this right away since the tests pass and the changes are purely mechanical.

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [submodule "llvm"]
 	path = llvm
-	url = ../../circt/llvm.git
+	url = git@github.com:circt/llvm.git
```
Comment (mikeurbach) [inline .gitmodules:3]: I know there was some discussion around this convention... @GeorgeLyon initiated the discussion in https://github.com/llvm/circt/pull/188 and we seemed to converge on the decision to use a relative path till we hit any issues.

@fabianschuiki is there some specific issue we hit here? I guess it assumes you are putting the copy of LLVM in a certain place that might not always hold. In my case, I don't even use the submodule, and prefer to keep a copy of LLVM in a very specific location I manage with CMake.

**Review by mikeurbach — COMMENTED** [review]
Thanks for handling the generated dialect files and the other misc. fixes.

File: .gitmodules (hunk 1 above)
Comment (fabianschuiki) [reply .gitmodules:3]: Damn, thanks for pointing this out. This is an oversight on my side. I fixed this locally and just committed this without really thinking about it. Will revert this change; #188 makes total sense and changing that decision should at least go through review.

Comment (fabianschuiki) [reply .gitmodules:3]: Reverted in 0797984c. Thanks for bringing this to my attention! :+1:

Comment (mikeurbach) [reply .gitmodules:3]: :+1: thanks!

**Comment by lattner** [comment]
Thank you for doing this Fabian!

**Pull request merged**